src/ioctls/vm.rs

```diff
@@ -193,6 +193,55 @@ impl VmFd {
         }
     }
 
+    /// Directly injects a MSI message as per the `KVM_SIGNAL_MSI` ioctl.
+    /// As per KVM API documentation this ioctl returns >0 on delivery,
+    /// 0 if guest blocked the MSI, and -1 on error.
```

Comment:
In our case it doesn't return -1 on error, but std::io::Error which contains the last error number. Since this is what happens to all ioctls, I wouldn't mention the error case. You can say something like: on success it return > 0 when the MSI is successfully delivered and 0 when the guest blocked the MSI.

Reply:
done